pkg/controller/runnable/reconciler.go

```diff
@@ -55,23 +58,63 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, err
 	}
 
-	condition, outputs, stampedObject := r.Realizer.Realize(ctx, runnable, logger, r.Repo)
+	if runnable == nil {
```

Comment:
maybe unrelated, but i noticed that in https://github.com/vmware-tanzu/cartographer/pull/358 we moved towards checking for `notfound` in the repository itself rather than at this level - should we include a change for that here too?

with the current code, I think this check for `nil` wouldn't make sense (unless we do the `if isnotfound()..` there in repository)

Reply:
ya this is just a weird diff i believe